### PR TITLE
2.2.25: adding friends-who-kept hover over keep button

### PR DESCRIPTION
**Pull request by 2is10**
also:
- hooking up real unread message/comment counts
- factoring out the optional measurement of the hover before showing it

### CODE REVIEW

File: packrat/scripts/google_inject.js [hunk 1]
```diff
@@ -485,7 +479,7 @@ api.log("[google_inject]");
 
     hit.count = hit.count - hit.users.length - (hit.isMyBookmark ? 1 : 0);
 
-    // Awful decision tree for clean text. Come up with a better way.
+    // Awful decision tree. Got a better way?
```
Comment (andrewconner) [inline packrat/scripts/google_inject.js:482]: :+1:

File: packrat/scripts/slider2.js [hunk 2]
```diff
@@ -255,6 +272,38 @@ slider2 = function() {
     var $html = $("html").removeClass("kifi-with-pane");
   }
 
+  function formatCountHtml(kept, isPrivate, numFriends, numTotal) {
+    var numOthers = (numTotal || 0) - (numFriends || 0) - (kept && !isPrivate ? 1 : 0);
+    // Awful decision tree. Got a better way?
+    if (kept) {
+      var priv = ""; // isPrivate ? " <span class=kifi-slider2-private>Private</span>" : "";
```
Comment (andrewconner) [inline packrat/scripts/slider2.js:279]: Could this be cleaned up since it's not used anymore for the ⓜⓔⓣⓡⓞ slider?

Comment (eishay) [inline packrat/scripts/slider2.js:279]: lol

Comment (2is10) [inline packrat/scripts/slider2.js:279]: I'll delete soon if we decide not to show. Need to ask Danny.

**Pull request merged**